code/modules/mining/equipment/resonator.dm

```diff
@@ -114,3 +114,10 @@
 	. = ..()
 	transform = matrix()*1.5
 	animate(src, transform = matrix()*0.1, alpha = 50, time = 4)
+
+/obj/effect/resonance/proc/replicate(turf/M, creator, timetoburst) 
+	for(var/turf/closed/T in orange(1, M))
+		if(istype(T, M))
+			var/turf/closed/mineral/T2 = T
+			if(T2.mineralType) // so we don't end up in the ultimate chain reaction - Super322
```

Comment:
You never check to see if it's `== M.mineraltype` despite your PR description.

Reply:
Also this entire block has some weird typecasting for what you want to do

Reply:
It's a port from pre-rebase Yogstation. I did not get the time to actually work on this to check for compatibility and with many bugs occurring with my personal desktop that prevented me from doing actual checking (on top of a general lack of experience) ;
I took the lazy man's route.

Reply:
Yogs code 👌

Reply:
So can I red pen your code or should I wait until you get it fixed?

Reply:
Yeah, go ahead.

Reply:
`T2.mineralType == M.mineralType` else this reacts with any nearby mineral wall.